Windows/CommonDialog/DataModel.cs

```diff
@@ -0,0 +1,109 @@
+﻿using Microsoft.Win32;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CommonDialogs
+{
+    class DataModel : INotifyPropertyChanged
```

Comment:
You never change any properties, you always replace the DataContext with a new instance. In that case there is no reason for `INotifyPropertyChanged`.